imgee/storage.py

```diff
@@ -263,32 +272,11 @@ def clean_local_cache(expiry=24):
     return n
 
 
-def wait_for_asynctasks(stored_file):
-    registry = imgee.registry
-
-    if not registry.connection:
-        return
-
-    # wait for upload to be complete, if any.
-    taskid = get_taskid('save_on_s3', stored_file.name+stored_file.extn)
-    if taskid in registry:
-        AsyncResult(taskid).get()
-
-    # wait for all resizes to be complete, if any.
-    s = get_taskid('resize_and_save', stored_file.name)
-    for taskid in registry.keys_starting_with(s):
-        AsyncResult(taskid).get()
-
-
-@celery.task(name='imgee.storage.delete', base=BaseTask)
-def delete(stored_file):
+def delete(stored_file, commit=True):
     """
     Delete all the thumbnails and images associated with a file, from local cache and S3.
     Wait for the upload/resize to complete if queued for the same image.
```

Comment:
The waiting was there with celery where this function will just wait if a resize operation is going on. Right now there is no wait. The delete function will delete everything even if something is going on parallel. What should be the ideal flow of this? Let's saw a resize task is going on. There is an entry for the file in taskregistry. what should the delete function do?